bin/faucet/src/client.rs

```diff
@@ -0,0 +1,353 @@
+}
+
+impl FaucetDataStore {
+    pub fn new(
+        faucet_account: Rc<RefCell<Account>>,
+        seed: Word,
+        root_block_header: BlockHeader,
+        root_chain_mmr: ChainMmr,
+    ) -> Self {
+        Self {
+            faucet_account,
+            seed,
+            block_header: root_block_header,
+            chain_mmr: root_chain_mmr,
+        }
+    }
+
+    fn update_faucet_account(&mut self, delta: &AccountDelta) -> Result<(), FaucetError> {
+        self.faucet_account
+            .borrow_mut()
+            .apply_delta(delta)
+            .map_err(|err| FaucetError::InternalServerError(err.to_string()))
+    }
+}
+
+impl DataStore for FaucetDataStore {
+    fn get_transaction_inputs(
+        &self,
+        account_id: AccountId,
+        _block_ref: u32,
+        _notes: &[NoteId],
+    ) -> Result<TransactionInputs, DataStoreError> {
+        let account = self.faucet_account.borrow();
+        if account_id != account.id() {
+            return Err(DataStoreError::AccountNotFound(account_id));
+        }
+
+        let empty_input_notes =
+            InputNotes::new(Vec::new()).map_err(DataStoreError::InvalidTransactionInput)?;
+
+        TransactionInputs::new(
+            account.clone(),
+            account.is_new().then_some(self.seed),
+            self.block_header,
+            self.chain_mmr.clone(),
+            empty_input_notes,
+        )
+        .map_err(DataStoreError::InvalidTransactionInput)
+    }
+
+    fn get_account_code(&self, account_id: AccountId) -> Result<ModuleAst, DataStoreError> {
+        let account = self.faucet_account.borrow();
+        if account_id != account.id() {
+            return Err(DataStoreError::AccountNotFound(account_id));
+        }
+
+        let module_ast = account.code().module().clone();
+        Ok(module_ast)
+    }
+}
+
+struct FaucetAuthenticator {
+    secret_key: AuthSecretKey,
+    rng: RefCell<ChaCha20Rng>,
+}
+
+impl FaucetAuthenticator {
+    pub fn new(secret_key: SecretKey) -> Self {
+        Self {
+            secret_key: AuthSecretKey::RpoFalcon512(secret_key),
+            rng: RefCell::new(ChaCha20Rng::from_entropy()),
+        }
+    }
+}
+
+impl TransactionAuthenticator for FaucetAuthenticator {
+    fn get_signature(
+        &self,
+        _pub_key: Word,
+        message: Word,
+        _account_delta: &AccountDelta,
+    ) -> Result<Vec<Felt>, AuthenticationError> {
+        let mut rng = self.rng.borrow_mut();
+        let AuthSecretKey::RpoFalcon512(k) = &self.secret_key;
+        get_falcon_signature(k, message, &mut rng)
+    }
+}
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+fn build_account(config: FaucetConfig) -> Result<(Account, Word, SecretKey), FaucetError> {
+    let token_symbol = TokenSymbol::new(config.token_symbol.as_str())
+        .map_err(|err| FaucetError::AccountCreationError(err.to_string()))?;
+
+    let seed: [u8; 32] = [0; 32];
```

Comment:
Should this be configurable somehow? Maybe not for this PR since it was like this before anyway

Reply:
Like storing the account seed in the config file?

Reply:
Yeah, although it's likely unnecessary for now. Let's defer until it's needed.